File: src/my_robot_tutorials/src/number_publisher.cpp
#include <random>
#include  <ros/ros.h>
#include <std_msgs/Int64.h>


int main(int argc, char **argv) 
{
    // Initialize random number generator
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<int> dis(0, 20);

    // REgister Node. The node is anonymous
    ros::init(argc, argv, "number_publisher", ros::init_options::AnonymousName);
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<std_msgs::Int64>("/number", 10);

    ROS_INFO("Publishing numbers from a node written with C++");

    // Get the frequency form ROS params
    double publish_frequency;
    // Get the ROS param and pack it in the variable publish_frequency
    nh.getParam("/number_publish_frequency", publish_frequency);

    nh.setParam("/a_param_from_cpp", "ztx");
    
    ros::Rate rate(3);

    while(ros::ok()) {
        // Generate random number
        int random_number = dis(gen);
        std_msgs::Int64 msg;        
        msg.data = random_number;
        pub.publish(msg);
        rate.sleep();
    }
}

File: src/my_robot_tutorials/scripts/number_publisher.py
#!/usr/bin/env python3

import random
import rospy
from std_msgs.msg import Int64

if __name__ == "__main__":
    # anonymous will add a random number after the name of the node. Therefore we can run multiple 
    # instances of this node
    rospy.init_node("number_publisher", anonymous=True)

    pub = rospy.Publisher("/number", Int64, queue_size=10)

    publish_frequency = rospy.get_param("/number_publish_frequency")
    rate = rospy.Rate(publish_frequency)

    rospy.set_param("/another_param_from_python", 3)

    while not rospy.is_shutdown():
        msg = Int64()
        msg.data = random.randint(0, 20)
        pub.publish(msg)
        rate.sleep()

    rospy.loginfo("Node was stopped")
    